OpenRA.Test/Fakes.cs

```diff
@@ -0,0 +1,214 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using OpenRA.Mods.Common.Traits;
+
+namespace OpenRA.Test
+{
+	public class FakeActor : IActor
+	{
+		private IWorld world;
+
+		public ActorInfo Info
+		{
+			get { throw new NotImplementedException(); }
+		}
+
+		public IWorld World
+		{
+			get { return world; }
+		}
+
+		public uint ActorID
+		{
+			get { return 1; }
+		}
+
+		public Player Owner
+		{
+			get
+			{
+				return null;
+			}
+
+			set
+			{
+			}
+		}
+
+		public T TraitOrDefault<T>()
+		{
+			return default(T);
+		}
+
+		public T Trait<T>()
+		{
+			return default(T);
+		}
+
+		public IEnumerable<T> TraitsImplementing<T>()
+		{
+			throw new NotImplementedException();
+		}
+
+		public T TraitInfo<T>()
+		{
+			return default(T);
+		}
+
+		public IEnumerable<Graphics.IRenderable> Render(Graphics.WorldRenderer wr)
+		{
+			throw new NotImplementedException();
+		}
+
+		public FakeActor(IWorld world)
+		{
+			// TODO: Complete member initialization
+			this.world = world;
+		}
+
+		public FakeActor()
+		{
+			// TODO: Complete member initialization
+		}
+	}
+
+	public class FakeWorld : IWorld
+	{
+		private FakeActor worldactor;
+		private IMap map;
+
+		public IActor WorldActor
+		{
+			get { return worldactor; }
+		}
+
+		public int WorldTick
+		{
+			get { throw new NotImplementedException(); }
+		}
+
+		public IMap Map
+		{
+			get { return map; }
+		}
+
+		public TileSet TileSet
+		{
+			get { throw new NotImplementedException(); }
+		}
+
+		public FakeWorld(IMap map)
+		{
+			// TODO: Complete member initialization
+			this.map = map;
+		}
+
+		public FakeWorld(IMap map, FakeActor worldactor)
+		{
+			// TODO: Complete member initialization
+			this.map = map;
+			this.worldactor = worldactor;
+		}
+	}
+
+	public class FakeMobileInfo : IMobileInfo
+	{
+		private Func<CPos, bool> conditions;
+
+		public int MovementCostForCell(World world, CPos cell)
+		{
+			if (conditions(cell))
+				return 125;
+			return int.MaxValue;
+		}
+
+		public bool CanEnterCell(World world, Actor self, CPos cell, out int movementCost, Actor ignoreActor = null, CellConditions check = CellConditions.All)
+		{
+			movementCost = MovementCostForCell(world, cell);
+			return conditions(cell);
+		}
+
+		public int GetMovementClass(TileSet tileset)
+		{
+			throw new NotImplementedException();
```

Comment:
This might be obvious if you are familiar with the code, but such an exception should provide a reason why isn't this method implemented.
Unless this is supposed to be implemented but still not finished?